src/dataprotection/azext_dataprotection/manual/custom.py

```diff
@@ -8,137 +8,40 @@
 # pylint: disable=too-many-locals
 # pylint: disable=line-too-long
 # pylint: disable=too-many-branches
+# pylint: disable=protected-access
 import uuid
 import re
 import time
 from knack.util import CLIError
 from knack.log import get_logger
-from azure.cli.core.util import sdk_no_wait
 from azext_dataprotection.vendored_sdks.resourcegraph.models import \
     QueryRequest, QueryRequestOptions
 from azext_dataprotection.manual import backupcenter_helper, helpers as helper
-# from azext_dataprotection.aaz.latest.data_protection.resource_guard import Show as ResourceGuardShow
 
 logger = get_logger(__name__)
 
 
-def dataprotection_resource_guard_list(client, resource_group_name=None):
-    if resource_group_name is not None:
-        return client.get_resources_in_resource_group(resource_group_name=resource_group_name)
-    return client.get_resources_in_subscription()
-
-
-def resource_guard_list_protected_operations(client, resource_group_name, resource_guards_name, resource_type):
-    resource_guard_object = client.get(resource_group_name, resource_guards_name)
-    protected_operations = resource_guard_object.properties.resource_guard_operations
+def resource_guard_list_protected_operations(cmd, resource_group_name, resource_guard_name, resource_type):
+    from azext_dataprotection.aaz.latest.dataprotection.resource_guard import Show as ResourceGuardShow
+    resource_guard_object = ResourceGuardShow(cli_ctx=cmd.cli_ctx)(command_args={
+        "resource_group": resource_group_name,
+        "resource_guard_name": resource_guard_name,
+    })
+    if resource_guard_object.get('properties'):
+        protected_operations = resource_guard_object.get('properties').get('resourceGuardOperations')
+    else:
+        raise CLIError("Error: Could not fetch resource guard.")
```

Comment:
`CLIError` is not recommended, please use the error types in this [doc]( https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md)

Reply:
Fixed it here and in other commands as well.